turbinia/workers/photorec.py

```diff
@@ -44,23 +42,32 @@ def run(self, evidence, result):
     # Create the new Evidence object that will be generated by this Task.
     output_evidence = PhotorecOutput()
     # Create a path that we can write the new file to.
-    base_name = os.path.basename(evidence.local_path)
     output_file_path = os.path.join(self.output_dir, 'photorec_output')
     photorec_log = os.path.join(self.output_dir, 'photorec.log')
     # Add the output path to the evidence so we can automatically save it
     # later.
-    output_evidence.local_path = output_file_path
+    output_evidence.local_path = ''.join((output_file_path, '.1'))
+    output_evidence.uncompressed_directory = output_evidence.local_path
     try:
       # Generate the command we want to run.
-      cmd = 'photorec /log /d {0:s} /cmd {1:s}  options,paranoid,keep_corrupted_fille,search'.format(
-          output_file_path, evidence.local_path)
+      cmd = ['photorec', '/log', '/d', output_file_path, '/cmd']
+      if evidence.device_path:
+        sudo = ['sudo']
+        sudo.extend(cmd)
+        cmd = sudo
+        cmd.append(evidence.device_path)
+      else:
+        cmd.append(evidence.local_path)
+      cmd.append('options,paranoid,keep_corrupted_file,search')
+      cmd = ' '.join(cmd)
       # Add a log line to the result that will be returned.
       result.log('Running photorec as [{0:s}]'.format(cmd))
       # Actually execute the binary
       self.execute(
           cmd, result, log_files=[photorec_log], new_evidence=[output_evidence],
           shell=True)
-      output_evidence.compress()
+      if os.path.exists(output_evidence.local_path):
+        output_evidence.compress()
```

Comment:
We should probably add a message here indicating that there was no output, and we might want to update the status message below to indicate that as well.